src/stactools/sentinel2/mgrs.py

```diff
@@ -0,0 +1,210 @@
+
+    def __init__(self, item: pystac.Item):
+        self.item = item
+        self.properties = item.properties
+
+    def __repr__(self) -> str:
+        return "<ItemMgrsExtension Item id={}>".format(self.item.id)
+
+    def apply(
+        self,
+        latitude_band: str,
+        grid_square: str,
+        utm_zone: Optional[int] = None,
+    ) -> None:
+        """Applies MGRS extension properties to the extended Item.
+
+        Args:
+            latitude_band : REQUIRED. The latitude band of the Item's centroid.
+            grid_square : REQUIRED. MGRS grid square of the Item's centroid.
+            utm_zone : The UTM Zone of the Item centroid.
+        """
+        self.latitude_band = validated_latitude_band(latitude_band)
+        self.grid_square = validated_grid_square(grid_square)
+        self.utm_zone = validated_utm_zone(utm_zone)
+
+    @property
+    def latitude_band(self) -> Optional[str]:
+        """Get or sets the latitude band of the datasource.
+        """
+        return self._get_property(LATITUDE_BAND_PROP, str)
+
+    @latitude_band.setter
+    def latitude_band(self, v: str) -> None:
+        self._set_property(LATITUDE_BAND_PROP,
+                           validated_latitude_band(v),
+                           pop_if_none=False)
+
+    @property
+    def grid_square(self) -> Optional[str]:
+        """Get or sets the latitude band of the datasource.
+        """
+        return self._get_property(GRID_SQUARE_PROP, str)
+
+    @grid_square.setter
+    def grid_square(self, v: str) -> None:
+        self._set_property(GRID_SQUARE_PROP,
+                           validated_grid_square(v),
+                           pop_if_none=False)
+
+    @property
+    def utm_zone(self) -> Optional[int]:
+        """Get or sets the latitude band of the datasource.
+        """
+        return self._get_property(UTM_ZONE_PROP, int)
+
+    @utm_zone.setter
+    def utm_zone(self, v: Optional[int]) -> None:
+        self._set_property(UTM_ZONE_PROP,
+                           validated_utm_zone(v),
+                           pop_if_none=False)
+
+    @classmethod
+    def get_schema_uri(cls) -> str:
+        return SCHEMA_URI
+
+    @classmethod
+    def ext(cls,
+            obj: pystac.Item,
+            add_if_missing: bool = False) -> "MgrsExtension":
+        """Extends the given STAC Object with properties from the :stac-ext:`MGRS
+        Extension <mgrs>`.
+
+        This extension can be applied to instances of :class:`~pystac.Item`.
+
+        Raises:
+
+            pystac.ExtensionTypeError : If an invalid object type is passed.
+        """
+        if isinstance(obj, pystac.Item):
+            cls.validate_has_extension(obj, add_if_missing)
+            return cast(MgrsExtension, MgrsExtension(obj))
+        else:
+            raise pystac.ExtensionTypeError(
+                f"MGRS Extension does not apply to type '{type(obj).__name__}'"
+            )
+
+    # @classmethod
```

Comment:
I left this in here commented in case we want to add summaries, will remove if we don't.

Reply:
Summaries would be good.

I supposed we'll want to get a PR for the mgrs extension back into PySTAC at some point as well.